catalog/spatial/geocoding/spatial-geocoding-lucene/src/test/java/org/codice/ddf/spatial/geocoding/query/GazetteerQueryCatalogTest.java

```diff
@@ -0,0 +1,324 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.geocoding.query;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.data.impl.types.LocationAttributes;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Location;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.codice.ddf.spatial.geocoding.GeoEntry;
+import org.codice.ddf.spatial.geocoding.GeoEntryMetacardType;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryException;
+import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GazetteerQueryCatalogTest {
+
+  private static final int RADIUS_IN_KM = 100;
+
+  private static final int MAX_RESULTS = 10;
+
+  private static final String BOSTON_WKT = "POINT (-71.0595703125 42.35771940022451)";
+
+  private static final String NEAR_BOSTON_WKT = "POINT(-71.07124328613281 42.353671973455704)";
+
+  private static final String BAD_WKT = "badWkt";
+
+  private static final String QUERY_STRING = "queryString";
+
+  private static final String BOSTON = "Boston";
+
+  private static final String USA_COUNTRY_CODE = "USA";
+
+  private static final MetacardType GEO_ENTRY_METACARD_TYPE =
```

Comment:
Should we be using our test libraries for creating the MetacardType?

Reply:
What testing library?

Reply:
I was thinking about https://github.com/connexta/gsr-yorktown/blob/master/catalog/core/common/catalog-core-common-xmlutils/src/test/java/mil/nga/gsr/catalog/xml/utils/InputTransformerTest.java#L41, but that would have to be refactored out of InputTransformerTest